lib/cea/cea608_memory.js

```diff
@@ -0,0 +1,513 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.cea.Cea608Memory');
+goog.provide('shaka.cea.Cea608Char');
+
+/**
+ * CEA-608 captions memory/buffer.
+ */
+shaka.cea.Cea608Memory = class {
+  /**
+   * @param {!shaka.cea.CeaDecoder} decoder CEA-608 decoder.
+   * @param {!number} fieldNum Field number.
+   * @param {!number} chanNum Channel number.
+   * @param {!boolean} textmode Indicates whether text mode is on/off.
+   */
+  constructor(decoder, fieldNum, chanNum, textmode) {
+    /**
+     * Buffer for storing decoded characters.
+     * @private @const {!Array<!Array<!shaka.cea.Cea608Char>>}
+     */
+    this.rows_ = [];
+
+    /**
+     * Current row.
+     * @private {!number}
+     */
+    this.row_ = 1;
+
+    /**
+     * Number of rows in the scroll window. Used for rollup mode.
+     * @private {!number}
+     */
+    this.scrollRows_ = 0;
+
+    /**
+     * Field number.
+     * @private {!number}
+     */
+    this.fldnum_ = fieldNum;
```

Comment:
Please give this a clearer name.  fieldNum was pretty good, so why make it into an unreadable fldnum here?